### PR TITLE
Test/i2c

### PR DESCRIPTION
**Pull request by electron101**
## Summary
On the Nucleo-L152RE board there are two I2C, I2C1 and I2C2 devices. But in the code there is no GPIO definition for I2C1. Because of this, an error occurred:

`chip/stm32_i2c.c:363:17: error: 'GPIO_I2C1_SCL' undeclared here (not in a function); did you mean 'GPIO_I2C2_SCL'?`
`chip/stm32_i2c.c:364:17: error: 'GPIO_I2C1_SDA' undeclared here (not in a function); did you mean 'GPIO_I2C2_SDA'?`

This error caused due to missing pin definition and fixed through first commit on this PR.
Although, build is success through this fix but there is no /dev/i2c1 device due to missing initializing and register process. So I added registration logic i2c device in other commits.

## Impact
###Add GPIO_I2C3_SDA, GPIO_I2C3_SCL macro.
###Added code for init and register process to I2C.

### CODE REVIEW

File: boards/arm/stm32/nucleo-l152re/include/board.h [hunk 1]
```diff
@@ -221,4 +221,11 @@
 #define GPIO_SPI1_MISO   GPIO_SPI1_MISO_2
 #define GPIO_SPI1_SCK    GPIO_SPI1_SCK_1
 
+/* I2C1 */
+
+#define GPIO_I2C1_SCL    GPIO_I2C1_SCL_2  /* PB8  CN5 pin 10, D15 */
+#define GPIO_I2C1_SDA    GPIO_I2C1_SDA_2  /* PB9  CN5 pin 9, D14 */
+
+/* I2C2 */
+
```
Comment (acassis) [inline boards/arm/stm32/nucleo-l152re/include/board.h:230]: Since I2C2 doesn't have pin mux option, please remove this comment

**Review by hartmannathan — COMMENTED** [review]
On code reading this looks okay and it builds for me. I don't have hardware to test it.

File: boards/arm/stm32/nucleo-l152re/include/board.h [hunk 2]
```diff
@@ -226,6 +226,4 @@
 #define GPIO_I2C1_SCL    GPIO_I2C1_SCL_2  /* PB8  CN5 pin 10, D15 */
 #define GPIO_I2C1_SDA    GPIO_I2C1_SDA_2  /* PB9  CN5 pin 9, D14 */
 
-/* I2C2 */
-
```
Comment (xiaoxiang781216) [inline boards/arm/stm32/nucleo-l152re/include/board.h]: let's merge the last two patch to the previous related patch instead.

**Comment by electron101** [comment]
Do I need any other action to merge?

пн, 7 февр. 2022 г. в 21:02, Xiang Xiao ***@***.***>:

File: boards/arm/stm32/nucleo-l152re/include/board.h (hunk 2 above)
Comment (electron101) [reply boards/arm/stm32/nucleo-l152re/include/board.h]: Do I need any other action to merge?

File: boards/arm/stm32/nucleo-l152re/src/stm32_appinitialize.c [hunk 3]
```diff
@@ -26,11 +26,14 @@
 
 #include <sys/types.h>
 #include <syslog.h>
+#include <debug.h>
 
 #include <nuttx/board.h>
 #include <nuttx/fs/fs.h>
 #include <nuttx/leds/userled.h>
 
+#include <stm32_i2c.h>
```
Comment (pkarashchenko) [inline boards/arm/stm32/nucleo-l152re/src/stm32_appinitialize.c:35]:
```suggestion
#include "stm32_i2c.h"
```

File: boards/arm/stm32/nucleo-l152re/include/board.h (hunk 2 above)
Comment (xiaoxiang781216) [reply boards/arm/stm32/nucleo-l152re/include/board.h]: git rebase --interactive HEAD~5
reorder the patch and change pick to squash in the editor
and git push -f again your branch.

File: include/sys/sysinfo.h [hunk 4]
```diff
@@ -0,0 +1,78 @@
+/****************************************************************************
+ * include/sys/sysinfo.h
```
Comment (xiaoxiang781216) [inline include/sys/sysinfo.h:2]: why has this change?

Comment (electron101) [reply include/sys/sysinfo.h]: I accidentally polluted the commit history, I'll try to fix it now

File: net/netdev/netdev_lladdrsize.c [hunk 5]
```diff
@@ -115,16 +115,6 @@ int netdev_lladdrsize(FAR struct net_driver_s *dev)
 #endif
 
 #ifdef CONFIG_NET_6LOWPAN
-#ifdef CONFIG_WIRELESS_BLUETOOTH
```
Comment (xiaoxiang781216) [inline net/netdev/netdev_lladdrsize.c:118]: why

File: boards/arm/stm32/nucleo-l152re/src/stm32_appinitialize.c [hunk 6]
```diff
@@ -80,6 +83,52 @@
 int board_app_initialize(uintptr_t arg)
 {
   int ret;
+#ifdef CONFIG_STM32_I2C1
+  FAR struct i2c_master_s *g_i2c1;
```
Comment (acassis) [inline boards/arm/stm32/nucleo-l152re/src/stm32_appinitialize.c:87]: Variables starting with symbol '"g_*" means it is a global variable, I think this is not the case here. Please fix it.

File: boards/arm/stm32/nucleo-l152re/src/stm32_appinitialize.c [hunk 7]
```diff
@@ -80,6 +83,52 @@
...
+#endif
+#ifdef CONFIG_STM32_I2C2
+  FAR struct i2c_master_s *g_i2c2;
```
Comment (acassis) [inline boards/arm/stm32/nucleo-l152re/src/stm32_appinitialize.c:90]: ditto

File: boards/arm/stm32/nucleo-l152re/src/stm32_appinitialize.c (hunk 6 above)
Comment (electron101) [reply boards/arm/stm32/nucleo-l152re/src/stm32_appinitialize.c]: fix it.

File: boards/arm/stm32/nucleo-l152re/src/stm32_appinitialize.c (hunk 7 above)
Comment (electron101) [reply boards/arm/stm32/nucleo-l152re/src/stm32_appinitialize.c]: fix it to

File: boards/arm/stm32/nucleo-l152re/src/stm32_appinitialize.c (hunk 3 above)
Comment (electron101) [reply boards/arm/stm32/nucleo-l152re/src/stm32_appinitialize.c]: fix it

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**